app/src/main/java/com/example/sharingang/ReportFragment.kt

```diff
@@ -0,0 +1,107 @@
+    private lateinit var reportedUsername: String
+    @Inject
+    lateinit var userRepository: UserRepository
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentReportBinding.inflate(inflater, container, false)
+        reporterId = args.reporterId
+        reportedId = args.reportedId
+        reportedUsername = args.reportedName
+
+
+        val radioButtons = listOf(
+            binding.radioUsername, binding.radioProfilePicture, binding.radioItem, binding.radioOther
+        )
+        setupRadioButtons(radioButtons)
+        setupCancelButton()
+        setupButtonOk(radioButtons)
+        binding.textReportedUsername.text = "Reporting $reportedUsername"
+
+        return binding.root
+    }
+
+    private fun setupRadioButtons(radioButtons: List<RadioButton>) {
+        for(radioButton: RadioButton in radioButtons) {
+            radioButton.setOnCheckedChangeListener { _, _ ->
+                binding.buttonOk.isEnabled =
+                    binding.radioUsername.isChecked || binding.radioProfilePicture.isChecked ||
+                    binding.radioItem.isChecked || binding.radioOther.isChecked
+            }
+        }
+    }
+
+    private fun setupCancelButton() {
+        binding.buttonCancel.setOnClickListener { view ->
+            view.findNavController().navigate(
+                ReportFragmentDirections.actionReportFragmentToUserProfileFragment(reportedId)
+            )
+        }
+    }
+
+    private fun setupButtonOk(radioButtons: List<RadioButton>) {
+        binding.buttonOk.setOnClickListener { view ->
+            var reason = ""
+            for (radioButton: RadioButton in radioButtons) {
+                if (radioButton.isChecked) {
+                    reason = getName(radioButton)
+                }
+            }
+            val reportDescription = binding.reportDescription.text.toString()
+            lifecycleScope.launch(Dispatchers.IO) {
+                val reporterUser = userRepository.get(reporterId)
+                val reportedUser = userRepository.get(reportedId)
+                userRepository.report(
+                    reporterUser = reporterUser!!,
+                    reportedUser = reportedUser!!,
+                    reason = reason,
+                    description = reportDescription,
+                )
+            }
+            view.findNavController().navigate(
+                ReportFragmentDirections.actionReportFragmentToUserProfileFragment(
+                    reportedId
+                )
+            )
+        }
+    }
+
+    private fun getName(radioButton: RadioButton): String {
+        return when (radioButton) {
+            binding.radioUsername -> "Inappropriate Username"
+            binding.radioProfilePicture -> "Inappropriate Profile Picture"
+            binding.radioItem -> "Inappropriate Item"
+            binding.radioOther -> "Other"
```

Comment:
I'd add strings in xml/strings.xml, in case we, for some reason, decide to change the language for example